File: backend-dotnet/Backend/Application/DTO/User/CreateUserRequest.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace Backend.Application.DTO.User;

public record CreateUserRequest(
    [property: JsonPropertyName("name")]
    [Required(ErrorMessage = "Name is requiredUser retrieved successfully")]
    [StringLength(100, MinimumLength = 2, ErrorMessage = "Name must be between 2 and 100 charactersUser retrieved successfully")]
    string Name,

    [property: JsonPropertyName("email")]
    [Required(ErrorMessage = "Email is requiredUser retrieved successfully")]
    [EmailAddress(ErrorMessage = "Email is not validUser retrieved successfully")]
    [StringLength(150, ErrorMessage = "Email must not exceed 150 charactersUser retrieved successfully")]
    string Email,

    [property: JsonPropertyName("password")]
    [Required(ErrorMessage = "Password is requiredUser retrieved successfully")]
    [StringLength(72, MinimumLength = 8, ErrorMessage = "Password must be between 8 and 72 charactersUser retrieved successfully")]
    string Password
);

File: web-spring/src/main/java/io/github/agussyahrilmubarok/web/application/dto/user/CreateUserRequest.java
package io.github.agussyahrilmubarok.web.application.dto.user;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

public record CreateUserRequest(
        @NotBlank(message = "{user.name.notBlank}")
        @Size(min = 2, max = 100, message = "{user.name.size}")
        String name,

        @NotBlank(message = "{user.email.notBlank}")
        @Email(message = "{user.email.invalid}")
        @Size(max = 150, message = "{user.email.size}")
        String email,

        @NotBlank(message = "{user.password.notBlank}")
        @Size(min = 8, max = 72, message = "{user.password.size}")
        String password) {}
